### ISSUE TITLE
Improve documentation for dropping of values and method calls

### ISSUE THREAD

**Issue by yorickpeterse** [inko-lang/inko#658]
Documentation for how values are dropped is a bit scattered:

- https://docs.inko-lang.org/manual/main/getting-started/memory-management/ outlines the general memory management strategy
- https://docs.inko-lang.org/manual/main/getting-started/variables/ covers dropping of fields and values
- Dropping of temporary unmoved values as part of method call chains isn't documented anywhere
- Method calls aren't documented at all apart from their syntax

I think that for the variable and fields guide, everything drop related needs to be moved into the memory management page. The memory management guide should then probably have the following top-level sections: ownership, drops, benefits, and costs.

In other words, if you want to learn more about drops, their order, etc, that should all be easily found on a single page.

**Comment by yorickpeterse** [comment]
Looking at this again, I think in the current manual structure it's pretty decent as the memory management section is discussed _before_ variables and fields.
